### PR TITLE
[terra-tabs] Added Drag and Drop for tab menu

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

- Added drag and drop for tab menu.
- Updated styles for tab content panel

**What was changed:**

- Added drag and drop for tab menu.
- Updated styles for tab content panel

**Why it was changed:**

- Identified as a functional gap for mpage pass through
- Tab content styles needed to switched appropriately based on the variant (framework/workspace). Framework tabs do not contain additional padding on right & left.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8867 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-tabs/src/common-tabs/_HiddenTab.jsx [hunk 1]
```diff
@@ -148,6 +165,36 @@ const HiddenTab = ({
...
+            aria-describedby={responseId}
+          >
+            <VisuallyHiddenText aria-hidden id={responseId} text={onFocusResponse} />
+            <IconKnurling />
+            <div className={cx('checkbox')}>{isSelected ? <IconCheckmark /> : null}</div>
+            {showIcon && <div>{icon}</div>}
+            <div className={cx('label', { 'with-icon': showIcon })}>{label}</div>
+          </div>
+        )}
+      </Draggable>
```
Comment (supreethmr) [inline packages/terra-tabs/src/common-tabs/_HiddenTab.jsx:194]: Are we ensuring that Add button which gets added as part of different PR will be out of draggable context.

File: packages/terra-tabs/src/common-tabs/_Tabs.jsx [hunk 2]
```diff
@@ -66,6 +67,10 @@ const propTypes = {
   * Callback function triggered when tab is drag and dropped .
    */
   onTabOrderChange: PropTypes.func,
+  /**
+   * intl object programmatically imported through injectIntl from react-intl.
+   * */
```
Comment (supreethmr) [inline packages/terra-tabs/src/common-tabs/_Tabs.jsx:72]:
```suggestion
   */
```
shouldn't this be private prop ?

File: packages/terra-tabs/src/common-tabs/_HiddenTab.jsx (hunk 1 above)
Comment (saket2403) [reply packages/terra-tabs/src/common-tabs/_HiddenTab.jsx:194]: Can be achieved by adding `isDragDisabled={condition which is true for add tab only}`. Will add it in the PR which is merged later.

**Comment by supreethmr** [comment]
Padding styles seems to be off with drag and drop icon

Screenshot 2023-08-10 at 11 28 36 AM

File: packages/terra-framework-docs/src/terra-dev-site/test/tabs/Tabs/ControlledTabs.test.jsx [hunk 3]
```diff
@@ -59,6 +88,9 @@ class ControlledTabs extends React.Component {
           {tab1}
           {tab2}
           {tab3}
+          {tab4}
+          {tab5}
+          {tab6}
```
Comment (supreethmr) [inline packages/terra-framework-docs/src/terra-dev-site/test/tabs/Tabs/ControlledTabs.test.jsx:93]: Can we include tab with icon also here to validate them on WDIO for alignment and other style properties

**Comment by saket2403** [comment]
Resolved Comments [c255ad7](https://github.com/cerner/terra-framework/pull/1734/commits/c255ad7d85ff3960533f2c203efdac09fc46359e), [458d0b6](https://github.com/cerner/terra-framework/pull/1734/commits/458d0b6ad90e59699c373a8473ba79ca9a97cd96)

**Review by supreethmr — APPROVED** [review]
+1 with Fixing Padding style on Menu 
Screenshot 2023-08-10 at 7 07 50 PM

**Comment by saket2403** [comment]
> +1 with Fixing Padding style on Menu Screenshot 2023-08-10 at 7 07 50 PM

Discussed on teams: No changes reqd

**Comment by sdadn** [comment]
Was this UX/Accessibility reviewed?

**Comment by supreethmr** [comment]
> Was this UX/Accessibility reviewed?

+1 @saket2403 was this reviewed by Accessibility team ??

**Pull request merged**